x-pack/plugins/synthetics/public/legacy_uptime/components/monitor_management/manage_locations/hooks/use_locations_api.ts

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useFetcher } from '@kbn/observability-plugin/public';
+import { useState } from 'react';
+import { useKibana } from '@kbn/kibana-react-plugin/public';
+import { PrivateLocation } from '../../../../../../common/runtime_types';
+import {
+  setSyntheticsPrivateLocations,
+  getSyntheticsPrivateLocations,
+} from '../../../../state/private_locations/api';
+
+export const useLocationsAPI = () => {
+  const [formData, setFormData] = useState<PrivateLocation>();
+  const [deleteId, setDeleteId] = useState<string>();
+
+  const { savedObjects } = useKibana().services;
+
+  const { data: currentPrivateLocations, loading: fetchLoading } = useFetcher(() => {
+    if (!formData) return getSyntheticsPrivateLocations(savedObjects?.client!);
+    return Promise.resolve(null);
+  }, [formData, deleteId]);
+
+  const { loading } = useFetcher(async () => {
+    if (currentPrivateLocations && formData) {
+      const result = await setSyntheticsPrivateLocations(savedObjects?.client!, {
+        locations: [
+          ...(currentPrivateLocations ?? []),
+          { ...formData, id: String((currentPrivateLocations?.length ?? 0) + 1) },
```

Comment:
Won't this logic cause id collisions? For example, if I have 5 private locations with ids 1,2,3,4,5, then delete the first one with id 1, now you have `currentPrivateLocations.length` equal to 4 and the next id generated being 5, so you have two locations with id 5.